app/src/main/java/com/example/parkingsystemjava/database/ReservationDataBase.java

```diff
@@ -0,0 +1,42 @@
+package com.example.parkingsystemjava.database;
+
+import com.example.parkingsystemjava.ReservationObject.Reservation;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class ReservationDataBase {
+    private static ReservationDataBase reservationDataBase;
+    private final HashMap<String, ArrayList<Reservation>> reservationHashMap;
+
+    public ReservationDataBase() {
+        this.reservationHashMap = new HashMap<>();
+    }
+
+    public static ReservationDataBase getInstance() {
+        if (reservationDataBase == null) {
+            reservationDataBase = new ReservationDataBase();
+        }
+        return reservationDataBase;
+    }
+
+    public void addValues(Reservation reservation) {
+        String parkingLots = reservation.getParkingLots();
+        if (reservationHashMap.get(parkingLots) == null) {
+            ArrayList<Reservation> reservations = new ArrayList<>();
+            reservations.add(reservation);
+            reservationHashMap.put(parkingLots, reservations);
+        } else {
+            reservationHashMap.get(parkingLots).add(reservation);
+        }
+    }
+
+    public Reservation getReservation(String parkingLots, String pass) {
+        ArrayList<Reservation> listReservation = reservationHashMap.get(parkingLots);
+        for (Reservation r : listReservation) {
```

Comment:
try to use more representative names, no just r

Reply:
Done,changed